src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,33 @@
-import React from 'react';
+import { ChangeEvent } from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/filter';
+
+export const TodoFilter = () => {
+  const dispatch = useDispatch();
+  const { query, sort } = useAppSelector(state => state.filter);
+
+  const handleSortChange = ((event: ChangeEvent<HTMLSelectElement>) => {
+    dispatch(actions.sortChange(event.target.value));
+  });
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    dispatch(actions.queryAdd(event.target.value));
+  };
+
+  const handleQueryClear = () => {
+    dispatch(actions.queryClear());
+  };
 
-export const TodoFilter: React.FC = () => {
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
```

Comment:
The form submission event is missing. It should prevent the default form submission behavior. Add onSubmit={(event) => event.preventDefault()} to the form element.